velox/connectors/hive/tests/HivePartitionFunctionTest.cpp

```diff
@@ -83,3 +83,101 @@ TEST_F(HivePartitionFunctionTest, bool) {
   assertPartitions(values, 500, {0, 1, 0, 0, 1});
   assertPartitions(values, 997, {0, 1, 0, 0, 1});
 }
+
+TEST_F(HivePartitionFunctionTest, int8_t) {
+  auto values = vm_.flatVectorNullable<int8_t>(
+      {std::nullopt,
+       64,
+       std::numeric_limits<int8_t>::min(),
+       std::numeric_limits<int8_t>::max()});
+
+  assertPartitions(values, 1, {0, 0, 0, 0});
+  assertPartitions(values, 2, {0, 0, 0, 1});
+  assertPartitions(values, 500, {0, 64, 20, 127});
+  assertPartitions(values, 997, {0, 64, 355, 127});
+}
+
+TEST_F(HivePartitionFunctionTest, int16_t) {
+  auto values = vm_.flatVectorNullable<int16_t>(
+      {std::nullopt,
+       30'000,
+       std::numeric_limits<int16_t>::min(),
+       std::numeric_limits<int16_t>::max()});
+
+  assertPartitions(values, 1, {0, 0, 0, 0});
+  assertPartitions(values, 2, {0, 0, 0, 1});
+  assertPartitions(values, 500, {0, 0, 380, 267});
+  assertPartitions(values, 997, {0, 90, 616, 863});
+}
+
+TEST_F(HivePartitionFunctionTest, int32_t) {
+  auto values = vm_.flatVectorNullable<int32_t>(
+      {std::nullopt,
+       2'000'000'000,
+       std::numeric_limits<int32_t>::min(),
+       std::numeric_limits<int32_t>::max()});
+
+  assertPartitions(values, 1, {0, 0, 0, 0});
+  assertPartitions(values, 2, {0, 0, 0, 1});
+  assertPartitions(values, 500, {0, 0, 0, 147});
+  assertPartitions(values, 997, {0, 54, 0, 482});
+}
+
+TEST_F(HivePartitionFunctionTest, float) {
+  auto values = vm_.flatVectorNullable<float>(
+      {std::nullopt,
+       2'000'000'000.0,
+       std::numeric_limits<float>::min(),
+       std::numeric_limits<float>::max()});
+
+  assertPartitions(values, 1, {0, 0, 0, 0});
+  assertPartitions(values, 2, {0, 0, 0, 1});
+  assertPartitions(values, 500, {0, 348, 108, 39});
+  assertPartitions(values, 997, {0, 544, 847, 632});
+}
+
+TEST_F(HivePartitionFunctionTest, double) {
+  auto values = vm_.flatVectorNullable<double>(
+      {std::nullopt,
+       300'000'000'000.5,
+       std::numeric_limits<double>::min(),
+       std::numeric_limits<double>::max()});
+
+  assertPartitions(values, 1, {0, 0, 0, 0});
+  assertPartitions(values, 2, {0, 1, 0, 0});
+  assertPartitions(values, 500, {0, 349, 76, 76});
+  assertPartitions(values, 997, {0, 63, 729, 729});
+}
+
+TEST_F(HivePartitionFunctionTest, Timestamp) {
+  // TODO Fix flatVectorNullable to set Timestamp.
```

Comment:
Would you like to take care of this TODO in a separate PR?

Reply:
Sure, will do later.

Reply:
Hi @mbasmanova, seems #778 has already done fixing `flatVectorNullable`, I change the test of `varchar` and `timestamp` to use the initalizer_list form of the API and it works. I created a [PR](https://github.com/facebookincubator/velox/pull/1625) on this. Thanks.